src/app/charts/extra-params-components/threshold.component.ts

```diff
@@ -48,54 +50,62 @@ export class ThresholdComponent implements AfterViewInit, OnChanges {
 
-    createForm() {
-        this.thresholdForm = this.fb.group({
-            comparatorCtl: [this.defaultComparator, Validators.required],
-            thresholdCtl: [this.defaultThreshold, Validators.required],
-            thresholdUnitCtl: [this.defaultTemperatureUnit, Validators.required]
-        });
+    constructor(private fb: FormBuilder) {}
 
-        this.thresholdForm.valueChanges.debounceTime(1000).subscribe(form => {
-            this.thresholdParamSelected.emit({data: {
-                'event': event,
-                'threshold_comparator': form.comparatorCtl,
-                'threshold': form.thresholdCtl,
-                'threshold_units': form.thresholdUnitCtl
-            }});
-        });
+    ngOnChanges(changes: any) {
+        /* Ignore change detection:
+            - before initalization
+            - from extraParams @Input whose initial values are all we want
+        */
+        if (changes.label && this.thresholdForm) {
+            this.thresholdForm.reset({
+                thresholdUnitCtl: this.defaultUnit,
+                comparatorCtl: this.defaultComparator,
+                thresholdCtl: this.defaultThreshold
+            });
+        }
     }
 
-    constructor(private fb: FormBuilder) {
-        this.createForm();
+    ngOnInit() {
+        this.createForm(); // must create form on init instead of constructor to capture @Input values
     }
 
     ngAfterViewInit() {
         // Since valueChanges triggers initially before parent is ready, wait until
         // parent is ready here and trigger it to draw chart with extra parameters.
         this.thresholdParamSelected.emit({data: {
-            'event': null,
             'threshold_comparator': this.thresholdForm.controls.comparatorCtl.value,
             'threshold': this.thresholdForm.controls.thresholdCtl.value,
             'threshold_units': this.thresholdForm.controls.thresholdUnitCtl.value
         }});
     }
 
-    ngOnChanges(changes: any) {
-        // listen for the indicator label to be set before changing options and defaults, if needed
+    createForm() {
+        this.evaluateVariable();
+        this.thresholdForm = this.fb.group({
+            comparatorCtl: [this.extraParams.threshold_comparator || this.defaultComparator, Validators.required],
+            thresholdCtl: [this.extraParams.threshold || this.defaultThreshold, Validators.required],
+            thresholdUnitCtl: [this.extraParams.threshold_units || this.defaultUnit, Validators.required]
+        });
+
+        this.thresholdForm.valueChanges.debounceTime(700).subscribe(form => {
+            this.thresholdParamSelected.emit({data: {
+                'event': event,
+                'threshold_comparator': form.comparatorCtl,
+                'threshold': form.thresholdCtl,
+                'threshold_units': form.thresholdUnitCtl
+            }});
+        });
+    }
+
+    private evaluateVariable() {
```

Comment:
Since we need to know the default unit, maybe we should pass `Indicator` instead of `label` as an input, such that we could just pull the default unit from there rather than attempting to magically derive it based on substrings in the label.

Reply:
Hm upon review,`Indicator.default_units` doesn't necessarily coordinate to pr/temp unit (C,F,K,in,etc). It can be `days`for some indicators, so derivation is necessary in some form. I updated it to depend on `Indicator.variables,` which I think is more readable.

Reply:
Nice catch.